chain/jsonrpc/src/lib.rs

```diff
@@ -579,7 +585,9 @@ impl JsonRpcHandler {
                                 );
                             }
                         }
-                        break Err(err);
+                        if finality == TxExecutionStatus::None {
+                            break Err(err);
+                        }
```

Comment:
With the ability to provide finality to requests of methods `tx`, `EXPERIMENTAL_tx_status`, and `Final` as the default status value, the behavior changes in some cases. It may be considered as a breaking of backward compatibility.
In particular, I can imagine 2 situations:
1. When the tx was just sent to the network, we previously showed "Unknown transaction" error next ~3-6 seconds. With this change, we will wait these ~3-6 seconds and then return the existing tx.
2. When the tx never existed (let's say it's the tx from testnet, and the user tries to find in on mainnet), and the user provides any finality other than None (or leaves the field blanc and we take the default value), user will wait for 10 seconds and then see Timeout error. Previously, we showed "Unknown transaction" almost immediately.

My opinion here:
The first one seems more like an improvement to me. (@khorolets will be happy because I will accidentally improve the correctness of Read RPC with this change)
The second one seems rare enough. If someone has a bunch of random hashes and they need to find all the existing transactions, they may use finality `None` to speed up the process